ark/utils/data_utils.py

```diff
@@ -4,6 +4,74 @@
 import numpy as np
 import xarray as xr
 
+from ark import settings
+from ark.utils.misc_utils import verify_in_list
+
+
+def relabel_images(fovs, all_data, label_maps, cell_label_col='fovs',
+                   fov_col=settings.FOV_ID):
+    """ Translates cell-ID labeled images according to the clustering assignment.
+
+    Takes a list of fovs, and relabels each image (array) according to the assignment
+    of cell IDs to cluster label.
+
+    Args:
+        fovs (list):
+            List of fovs to relabel.
+        all_data (pandas.DataFrame):
+            data including fovs, cell labels, and cell expression matrix for all markers.
+        label_maps (xr.DataArray):
+            xarray of label maps for multiple fovs
+        cell_label_col (str):
+            column with the fov names in label_maps.
+        fov_col (str):
+            column with the fovs names in all_data.
+    Returns:
+        xr.DataArray:
+            The relabeled images (dims: ["fovs", "rows", "cols"]).
+    """
+
+    # check if included fovs found in fov_col
+    verify_in_list(fov_names=fovs, unique_fovs=all_data[fov_col].unique())
+    verify_in_list(fov_names=fovs, unique_fovs=label_maps[cell_label_col])
+
+    img_data = []
+    for fov in fovs:
+        df = all_data[all_data[fov_col] == fov]
+        labels_dict = dict(zip(df.label, df.cluster_labels))
+        img_data.append(
```

Comment:
I think this line is a bit too long, hard to see what's going on. I think it would be better to split it into one line that calls `relabel_img_array`, and a second line that appends the output.

Also, you can directly index into xarrays, which will simplify the expression:
`label_maps[fov, :, :, 0]`

Reply:
I split it to multiple lines to keep it more clear. I tried `label_maps[fov, :, :, 0]` (and some variants) and it didn't work, is it the exact syntax?

Reply:
No I'm an idiot, It's `label_maps.loc`. Sorry!